packages/tpetra/core/src/Tpetra_Import_Util2.hpp

```diff
@@ -712,6 +719,17 @@ sortAndMergeCrsEntries (const Teuchos::ArrayView<size_t> &CRS_rowptr,
   return sortAndMergeCrsEntries(CRS_rowptr, CRS_colind, CRS_vals);
 }
 
+template<class rowptr_view_type, class colind_view_type, class vals_view_type>
+void
+sortAndMergeCrsEntries(const rowptr_view_type& CRS_rowptr,
+		       const colind_view_type& CRS_colind,
+		       const vals_view_type& CRS_vals)
+{
+  using execution_space = typename vals_view_type::execution_space;
+  KokkosSparse::sort_crs_matrix<execution_space, rowptr_view_type,
```

Comment:
This function needs to sort as well as compress repeated entries right? ``sort_crs_matrix`` only does the sorting. Unfortunately this will probably take a little refactoring because ``sort_and_merge_crs_matrix`` has to reallocate entries/values, and so it returns a new matrix. But CRS_colind and CRS_vals are const here.

Reply:
@brian-kelley Yes, that's right.

Reply:
Brian, I am not so sure that CRS_colind and CRS_vals are const since they are getting resized only a line or two below in the Tpetra calling code?